### ISSUE TITLE
I have 2 issues with this compositor

### ISSUE THREAD

**Issue by AnErrupTion** [labwc/labwc#772]
Hello! I love this compositor, it's fast and lightweight, and thanks for actively developing it! However, I'm encountering 2 major issues when playing games:

1. When a game is in fullscreen (for example Minecraft or BeamNG.drive), the cursor image itself will appear on the screen but will be stuck, I can still move the mouse and select stuff but the cursor image itself won't move, it's frustating :/

2. In Minecraft specifically, when **not** in fullscreen, I get progressive input lag. What I mean is that, as time goes on, I'll get extremely bad input lag, so moving my mouse takes something like 2 seconds or more for the game to register. This actually happens very fast and within less than a minute of gameplay. This only seems to affect older versions of the game such as 1.8.9, I don't have issues with newer versions such as 1.19.3

I've tried to search issues in the repository for both of these issues and haven't found anything... any help is appreciated!

**Comment by Consolatis** [comment]
> When a game is in fullscreen (for example Minecraft or BeamNG.drive), the cursor image itself will appear on the screen but will be stuck, I can still move the mouse and select stuff but the cursor image itself won't move, it's frustating :/

see https://github.com/labwc/labwc/pull/733/files
tl;dr: until labwc implements the missing functionality, wrapping your game in gamescope should hopefully fix this.

> In Minecraft specifically, when not in fullscreen, I get progressive input lag. What I mean is that, as time goes on, I'll get extremely bad input lag, so moving my mouse takes something like 2 seconds or more for the game to register. This actually happens very fast and within less than a minute of gameplay. This only seems to affect older versions of the game such as 1.8.9, I don't have issues with newer versions such as 1.19.3

The same workaround could help with this, although I've never heard about increasing input lag before.
Do you use a high polling mouse by chance? If Minecraft doesn't process the incoming events fast enough I assume it could show those symptoms.

**Comment by Joshua-Ashton** [comment]
I'll try and take a look at the missing cursor stuff :)

**Comment by AnErrupTion** [comment]
> wrapping your game in gamescope should hopefully fix this.

Thanks, I'll try this!

> Do you use a high polling mouse by chance? If Minecraft doesn't process the incoming events fast enough I assume it could show those symptoms.

I use an MX Master 3, which by no means has a high polling rate. Also, this issue only happens with LabWC, on KWin (KDE Plasma) it works fine.

**Comment by AnErrupTion** [comment]
It seems like gamescope solved all my issues! Although, for some reason, using fullscreen gamescope is less smooth than using it windowed. I'm not sure how to describe it, but it's definitely less smooth in fullscreen. I'm not sure if this is a known thing or not...

**Comment by Consolatis** [comment]
Maybe you could do something like setting gamescope to your desktop resolution but keep it windowed, then toggle the labwc internal fullscreen function (`A-Space` menu -> Fullscreen or the `ToggleFullscreen` keybind).

Take this advice with a grain of salt though, I've never used gamescope personally.

**Comment by Consolatis** [comment]
As gamescope seems to have fixed the issue, I'll close this for now.

Feel free to share your experience regarding the fullscreen / windowed performance though (it might also help others who end up here searching for a solution) or open a new issue if you detect further issues.

**Comment by AnErrupTion** [comment]
> Maybe you could do something like setting gamescope to your desktop resolution but keep it windowed, then toggle the labwc internal fullscreen function (`A-Space` menu -> Fullscreen or the `ToggleFullscreen` keybind).
> 
> Take this advice with a grain of salt though, I've personally never used gamescope.

I'm not able to find the ``ToggleFullscreen`` keybind on labwc 0.6.1. Perhaps it's currently in master only? Also, when doing Alt + Space, the menu appears but I can't select anything because gamescope captures my mouse in the game, and I wasn't able to use the keyboard either to navigate the menu.

**Comment by Consolatis** [comment]
> I'm not able to find the ToggleFullscreen keybind on labwc 0.6.1. Perhaps it's currently in master only?

Its not a default keybind but it [exists](https://labwc.github.io/labwc-actions.5.html).
You can create a binding in your `~/.config/labwc/rc.xml` like so:
```xml
    <keybind key="W-f">
      <action name="ToggleFullscreen" />
    </keybind>
```
See [here](https://github.com/labwc/labwc/blob/master/docs/rc.xml) for an example in case you don't have a default config.
Depending on your distribution the same example file might also exist in `/usr/share/doc/labwc/`.

> Also, when doing Alt + Space, the menu appears but I can't select anything because gamescope captures my mouse in the game,

Heh, I didn't think about that. Makes sense.

**Comment by AnErrupTion** [comment]
Well, even with this, in fullscreen it's still not as smooth as in windowed mode. That's odd...

**Comment by Consolatis** [comment]
> That's odd...

Indeed. As I can't really test this myself (my daily driver is an Arm64 SOC) I am not sure what is going on.
If its really just the fullscreen part this could also work:
```xml
    <keybind key="W-m">
      <action name="ToggleDecorations" />
      <action name="ToggleMaximize" />
    </keybind>
```
That should result in somewhat of "borderless fullscreen". E.g. the application (gamescope in this case) doesn't know it is actually in fullscreen. But this is really guesswork now, maybe @Joshua-Ashton can shed a light on the performance difference between fullscreen and windowed mode.

**Comment by AnErrupTion** [comment]
Actually, this is even weirder, because it's also not as smooth as in windowed mode when i use the borderless windowed mode...
